build.gradle

```diff
@@ -21,6 +21,29 @@ buildscript {
             // 2.0.0-rc1-SNAPSHOT -> 2.0.0.0-rc1-SNAPSHOT (opensearch_build)
             opensearch_build += "-SNAPSHOT"
         }
+        getPrometheusBinaryLocation = { ->
+            def os = org.gradle.internal.os.OperatingSystem.current()
+            println "*** Building on ${os.familyName} / ${os.name} / ${os.version} / ${System.getProperty("os.arch")}."
+            // Currently only considering linux and macos. Need to evolve this logic once we start supporting other os types.
+            if (os.isMacOsX()) {
+                //x64, x86-64, amd64
+                if (System.getProperty("os.arch").startsWith("x") || System.getProperty("os.arch").startsWith("amd")) {
+                    return "https://github.com/prometheus/prometheus/releases/download/v2.37.2/prometheus-2.37.2.darwin-amd64.tar.gz"
```

Comment:
could you compose downlink using version, platform and arch. prometheus-{version}.{platform}-{arch}

Reply:
I have done that...I couldn't push the latest commit last night. 
There are too many names for the same arch type and OS family. So slightly simplified the logic.